sort/js/search.js

```diff
@@ -0,0 +1,34 @@
+var binarySearch = (function () {
+  var recursiveSearch = function (items, key, left, right) {
+    if (_isInvalid(items, key)) return;
+    if (left === undefined) left = 0;
+    if (right === undefined) right = items.length - 1;
+    var index = left + right >>> 1;
+    if (left > right) return -1;
+    if (items[index] === key) return index;
+    if (items[index] > key) {
+      return recursiveSearch(items, key, left, index - 1);
+    } else {
+      return recursiveSearch(items, key, index + 1, right);
+    }
+  };
+
+  var cyclicSearch = function (items, key) {
+    if (_isInvalid(items, key)) return;
```

Comment:
it's good if you understand why `_isInvalid` is defined here when you used function expression